tests/cypress/e2e/for-teacher_page/class_page/class-survey.cy.js

```diff
@@ -0,0 +1,82 @@
+import {loginForTeacher} from '../../tools/login/login.js'
+
+const surveyView = body => body.find("#survey")
+const classView = body => body.find(".view_class")
+describe('Class Survey View', () => {
+  it('Can respond to 1 questions', () => {
+    
+    loginForTeacher();
+
+    cy.get(".view_class").first().click()
+    cy.get("body").then(surveyView).then(survey => {
+        if (survey.length){
+            survey = cy.get("#survey")
+            survey.should("exist")
+            survey.get("#input").type("test")
+            survey.get("#submit").click()
+        }
+    })
+  })
+
+  it('Should have 3 questions remaning, can respond to those', () => {
+    
+    loginForTeacher();
+
+    cy.get(".view_class").first().click()
+    cy.get("body").then(surveyView).then(survey => {
```

Comment:
These tests are dependent on one another. This is not advisable unless on specific circumstances where we want to save some time, but we should avoid them as much as possible. Our philosophy with Cypress is making the tests independent from one another. See here: https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests#Test-Isolation